pkg/model/field.go

```diff
@@ -58,8 +59,18 @@ func newField(
 	if shapeRef != nil {
 		shape = shapeRef.Shape
 	}
-	if shape != nil {
+	if cfg != nil && len(cfg.Type) > 0 {
+		gt = cfg.Type
+		if strings.HasPrefix(gt, "[]") {
+			gte = strings.TrimPrefix(gt, "[]")
+			gtwp = gte
+		}
+	} else if shape != nil {
 		gte, gt, gtwp = cleanGoType(crd.sdkAPI, crd.cfg, shape)
+		if cfg != nil && cfg.Type == "list" &&
+			shape.Type != "list" {
+			gt = "[]" + gt
+		}
```

Comment:
In this function, if we see `fieldConfig.Type != nil`, we should just be bypassing all this logic, no? Regardless of whether there is a slice type or not...